src/cunumeric/matrix/solve_template.inl

```diff
@@ -0,0 +1,91 @@
+/* Copyright 2021-2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include <vector>
+
+#include "core/comm/coll.h"
+
+// Useful for IDEs
+#include "cunumeric/matrix/solve.h"
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <VariantKind KIND, LegateTypeCode CODE>
+struct SolveImplBody;
+
+template <LegateTypeCode CODE>
+struct support_solve : std::false_type {
+};
+template <>
+struct support_solve<LegateTypeCode::DOUBLE_LT> : std::true_type {
+};
+template <>
+struct support_solve<LegateTypeCode::FLOAT_LT> : std::true_type {
```

Comment:
I think the code doesn't support arrays of the `float16` type even though the python code accepts them. you should either modify the python code to convert them to `float32` and back, or do the conversion in the C++ code.

Reply:
https://github.com/nv-legate/cunumeric/pull/518#discussion_r945238447

Reply:
I was wrong about `float16` and NumPy doesn't seem to support it. Nevertheless, the current code doesn't handle the type error correctly. FYI, `np.dytpe(np.float16).kind == "f"`